Sources/AsyncAlgorithms/zip/Zip2StateMachine.swift

```diff
@@ -0,0 +1,519 @@
+//
+//  Zip2StateMachine.swift
+//  
+//
+//  Created by Thibault Wittemberg on 13/09/2022.
+//
+
+final class Zip2Runtime<Base1: AsyncSequence, Base2: AsyncSequence>
```

Comment:
previously we didn't need a reference type - what is the reasoning here?

Reply:
I've done the same as @FranzBusch in the MergeStorage I think. I've seen that it ease the usage of `self.` inside the concurrent code of the tasks. I've tried to make it a struct (which is doable) but I don't see the gain here ?